crates/nu-command/src/filesystem/rm.rs

```diff
@@ -381,11 +381,24 @@ fn rm(
                         {
                             unreachable!()
                         }
-                    } else if metadata.is_file()
-                        || is_socket
-                        || is_fifo
-                        || metadata.file_type().is_symlink()
-                    {
+                    } else if metadata.is_symlink() {
+                        // In Windows, symlink pointing to a directory can be removed using
+                        // std::fs::remove_dir instead of std::fs::remove_file.
+                        #[cfg(windows)]
+                        match f.metadata() {
```

Comment:
What about using `and_then`?
```rs
f.metadata().and_then(|metadata|
    if metadata.is_dir() { std::fs::remove_dir(&f) }
    else { std::fs::remove_file(&f) }
}?
```

Reply:
> What about using and_then?

Indeed, it's a better way.
I'm going to refactor it.